### ISSUE TITLE
Cannot link a nested object if it's called CLASS 

### ISSUE THREAD

**Issue by xeno-by** [scala-native/scala-native#1155]
```scala
trait C {
  def foo = "bar"
}

object C {
  object CLASS extends C
}

object Hello extends App {
  println(C.CLASS)
}
```

```
> run
[info] Compiling 1 Scala source to /Users/eburmako/Projects/tmp/target/scala-2.11/classes...
[info] Linking (1089 ms)
[error] cannot link: @C$CLASS$
[error] unable to link
[error] (compile:nativeLinkNIR) unable to link
[error] Total time: 1 s, completed Feb 1, 2018 2:13:04 PM
```

**Comment by densh** [comment]
This is a macOS-only issue, caused by case-insensitive filesystem.

**Comment by densh** [comment]
@sjrd Given symbol names `C$class` for `object C`and `C$CLASS` for `object CLASS extends C` (for a code snippet above), scalac emits first one with a dollar in the end and the second one without it, any idea where that logic is implemented?

**Comment by sjrd** [comment]
`C$class` without dollar is the implementation class of `trait C`. `C$CLASS` is the module class of `C.CLASS`, and it receives a dollar because all module classes receive a dollar. I don't know by heart for scalac, but for Scala.js it is implemented here: https://github.com/scala-js/scala-js/blob/ec5b328330276b9feb20cccadd75e19d27e887d3/compiler/src/main/scala/org/scalajs/nscplugin/JSEncoding.scala#L226-L240

**Comment by densh** [comment]
@sjrd Awesome, thanks! Here is a quick fix: #1156.
